drgn/helpers/linux/llist.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) 2022, Oracle and/or its affiliates.
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+"""
+Lock-less NULL terminated single linked lists
+------------
+
+The ``drgn.helpers.linux.llist`` module provides helpers for working with the
+lockless single linked list implementations (``struct llist_head`` and ``struct
+llist_node``) in :linux:`include/linux/llist.h`.
```

Comment:
Slight copy-editing:

```suggestion
Lockless Lists
--------------

The ``drgn.helpers.linux.llist`` module provides helpers for working with the
lockless, ``NULL``-terminated, singly-linked list implementation in
:linux:`include/linux/llist.h` (``struct llist_head`` and ``struct
llist_node``).
```